### PR TITLE
Feature 5: Buy Heart Use Case

### PR DESCRIPTION
**Pull request by harshsbajwa**
Implemented the Buy Heart Use Case according to feature, also implemented a test. 

BuyHeartUC checks for points and pet health, if pet health is max, do nothing, if user points are 0, do nothing, else buy hearts num of hearts and lose hearts num of points

If pet health is 0, then prevent user from buying hearts, since we agreed that when pet health reaches 0 the user would have to do something different from just buy hearts

Changed customization parameter in Pet for consistency with Customization entity

Deleted tutorial folder and HelloWorld

### CODE REVIEW

**Comment by XinyingXu** [comment]
I think this looks good for now.

**Review by XinyingXu — APPROVED** [review]
The use case looks good for now.

**Comment by BetterthanNothin** [comment]
src/test/java/useCases/BuyHeartUCTest.java
You can create a pet without an image icon now, as such, remove that as a parameter.

src/main/java/entities/Pet.java
The customization input for the constructor is not used. Why?

src/main/java/tutorial/HelloWorld.java
Don't delete it, we can still use it as a main.

File: src/main/java/entities/Customization.java [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 public class Customization {
 
+    public static ArrayList<Item> customizations;
     public boolean is_currently_equipped;
     public Item current_equipment;
```
Comment (stephaniesaab) [inline src/main/java/entities/Customization.java:12]: Could you add a Constructor JavaDoc here, simply stating what the default values the empty constructor will set is enough.

File: src/main/java/entities/Pet.java [hunk 2]
```diff
@@ -10,17 +10,23 @@ public class Pet {
     private int currHealth;
     private final String name;
     private final ImageIcon skin;
-    private final Customization[] customizations;
-    private Hat hat;
+    private final Customization customization;
```
Comment (stephaniesaab) [inline src/main/java/entities/Pet.java:14]: Please add some documentation about what each variable represents, just a single sentence is enough.

File: src/main/java/entities/User.java [hunk 3]
```diff
@@ -21,14 +20,16 @@ public int getPoints(){
...
-    public Checklist getTODOlist() {
-        return TODOlist;
-    }
-
-    public Checklist getDONElist() {
-        return DONElist;
-    }
+    
+    // TODO: UNCOMMENT WHEN TODOlist and DONElist entities are merged into main
+
```
Comment (stephaniesaab) [inline src/main/java/entities/User.java:25]: Please change the names of these from TODO and DONE to ToDoList and DoneList, to fit the java documentation standards

**Review by stephaniesaab — APPROVED** [review]
Hey Harsh, this code looks good. Just a few things to review are: 1) Could you please document each attribute in the Customization entity, and label them under '== Attributes ==' or something equivalent. I've also added a few comments that are mostly formatting issues, so we're standardized.

**Pull request merged**